Sources/StreamChat/Models/CurrentUser.swift

```diff
@@ -113,3 +113,76 @@ public class CurrentChatUser: ChatUser {
         $_mutedChannels = (mutedChannels, underlyingContext)
     }
 }
+
+/// The total unread information from the current user.
+public struct CurrentUserUnreads {
+    /// The total number of unread channels.
+    public var totalUnreadChannelsCount: Int
+    /// The total number of unread threads.
+    public var totalUnreadThreadsCount: Int
+    /// The unread information per channel.
+    public var unreadChannels: [CurrentUserUnreadChannel]
+    /// The unread information per thread.
+    public var unreadThreads: [CurrentUserUnreadThread]
+    /// The unread information per channel type.
+    public var unreadChannelsByType: [CurrentUserUnreadChannelByType]
+}
+
+/// The unread information of a channel.
+public struct CurrentUserUnreadChannel {
```

Comment:
Thinking if we really need the `CurrentUser` prefix in these models. How are they called in the REST API?

Reply:
They don't have any name, it is just a structure in the REST API, the property is called `unreadChannels`.

The `CurrentUser` prefix is to identify the call, and that these unreads are from the current user `/unread` endpoint. If we just name `UnreadChannel,`, I'm afraid it could clash with other future types 🤔 But I guess we can remove the prefix, not 100% sure about what to do here. But if you think it is fine I can rename it 👍

Reply:
I think it's better to rename it. Unread stuff is usually available only for the current user, don't think it would be used elsewhere.

Reply:
Will update it then 👍